apps/modernization-ui/src/apps/page-builder/pages/AddNewPage/AddNewPageFields.tsx

```diff
@@ -53,7 +54,13 @@ export const AddNewPageFields = (props: AddNewPageFieldProps) => {
                 name="name"
                 rules={{
                     required: { value: true, message: 'Name is required.' },
-                    ...validPageNameRule
+                    ...validPageNameRule,
+                    validate: (value) => {
+                        return (
+                            props.pageSummaries.some((page) => page.name === value) === false ||
+                            'Page name already exists'
+                        );
```

Comment:
Suggestion(Non-blocking): Maybe putting this in its own file the same way `validPageNameRule` is done might be better?